core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java

```diff
@@ -186,6 +190,46 @@ public void testCloudSqlInstanceForcesRefresh() throws Exception {
     assertThat(refreshCount.get()).isEqualTo(2);
   }
 
+  @Test
+  public void testCloudSqlInstanceRetriesOnInitialFailures() throws Exception {
+    InstanceData data =
+        new InstanceData(
+            null, new SslData(null, null, null), Instant.now().plus(1, ChronoUnit.HOURS));
+
+    AtomicInteger refreshCount = new AtomicInteger();
+    final PauseCondition refresh0 = new PauseCondition();
+
+    CloudSqlInstance instance =
+        new CloudSqlInstance(
+            "project:region:instance",
+            (instanceName, accessTokenSupplier, authType, executor, keyPair) -> {
+              int c = refreshCount.get();
+              refreshCount.incrementAndGet();
+              switch (c) {
+                case 0:
+                  refresh0.pause();
+                  throw new RuntimeException("bad request 0");
+                case 1:
+                  throw new RuntimeException("bad request 1");
```

Comment:
Why do we need two runtime exceptions? Would one suffice for this test?

Reply:
Good point. Fixed.